### DISCUSSION TITLE
How to test usage of notificationCenter

### DISCUSSION THREAD

**Discussion by martin-muller** [tgrapperon/swift-dependencies-additions#42] [Q&A]
Hi all,

I'm wondering what's the best approach to test that my client has sent a notification.

```swift
struct BestClient {
    let doImportantWork: () async throws -> Void
}

extension BestClient {
    static let live = {
        @Dependency(\.notificationCenter) var notificationCenter
        
        return BestClient(
            doImportantWork: {
                // ...
                
                notificationCenter.post(name: .init("important"))
            }
        )
    }()
}
```

My initial instinct was to override the `notificationCenter` dependency using `withDependencies` and either create a whole new `NotificationCenter.Dependency` or override just the `post` and assert that the call was made, but the init is internal and the funcs are internal lets. Then I thought I can import `DependenciesAdditions` as testable and have the init available, but weirdly enough that didn't work, maybe something in our setup. 

Im aware there is public init that takes `NotificationCenter`, but since we already have the wrapper client object `NotificationCenter.Dependency`, it seems like unnecessary complication to use the Foundation's `NotificationCenter` for testing.

Thank you for the wonderful library!

**Comment by tgrapperon** [comment]
Hey @martin-muller! The initial variant of `NotificationCenter` had a custom value that allowed some level of observation/introspection, but it was using a protocol, and I got rid of it after some time to simplify the API. I'll check if this is still easily achievable given the API that finally shipped!
